jQuery/src/index.js

```diff
@@ -0,0 +1,101 @@
+        },
+        onAppointmentUpdating: (e) => {
+            handleAppointmentActions(e, e.newData, popup);
+        }
+    })
+});
+
+const popupContentTemplate = function () {
+    return $('<div>').append(
+        $(`<p>There is a recurrent appointment in this cell.</p>`),
+    );
+};
+
+const popup = $('#popup').dxPopup({
+    contentTemplate: popupContentTemplate,
+    width: 300,
+    height: 150,
+    container: '.dx-viewport',
+    showTitle: true,
+    title: 'Information',
+    visible: false,
+    dragEnabled: false,
+    hideOnOutsideClick: true,
+    showCloseButton: false,
+    position: {
+        at: 'center',
+        my: 'center',
+    },
+    toolbarItems: [{
+        widget: 'dxButton',
+        toolbar: 'bottom',
+        location: 'after',
+        options: {
+            text: 'Close',
+            onClick() {
+                popup.hide();
+            },
+        },
+    }],
+}).dxPopup('instance');
+
+const formattedDate = (date) => new Date(date.getFullYear(), date.getMonth(), date.getDate());
+
+const handleAppointmentActions = (e, appointmentData, popup) => {
+    let startTime = appointmentData.startDate.getTime();
+    const endTime = appointmentData.endDate.getTime();
+    const recurringAppointment = defaultData.filter((appointment) => appointment?.recurrenceRule).map((appointment) => {
+        return {
+            ...appointment,
+            startDate: new Date(appointment.startDate),
+            endDate: new Date(appointment.endDate),
+        };
+    });
+
+    recurringAppointment.find((appointment) => {
+        const recurrenceOptions = rrule.rrulestr(appointment.recurrenceRule);
+        const rule = new rrule.RRule({
+            freq: recurrenceOptions.options.freq,
+            interval: recurrenceOptions.options.interval,
+            byweekday: recurrenceOptions.options.byweekday,
+            dtstart: appointment?.startDate,
+        })
+        const betweenDate = rule.between(e.component.getStartViewDate(), e.component.getEndViewDate())
+        const appointmentDuration = appointment.endDate.getTime() - appointment.startDate.getTime();
+
+        if (betweenDate.length > 0) {
+            betweenDate.find((date) => {
+                let recurrentStartTime = date.getTime();
+
+                if(appointment.allDay) {
+                    recurrentStartTime = recurrentStartTime - 86400000
+                    startTime = formattedDate(new Date(startTime)).getTime()
+                }
+                const recurrentEndTime = recurrentStartTime + appointmentDuration;
+
+                if (
+                    startTime === recurrentStartTime
+                    || startTime > recurrentStartTime && startTime < recurrentEndTime
+                    || endTime > recurrentStartTime && endTime <= recurrentEndTime
+                ) {
+                    e.cancel = true;
+                    popup.show();
+                }
+            })
+        }
+    })
+}
```

Comment:
Let's change our algorithm of detecting overlap between a new appointment and existing recurrent appointments:

**1) Let's move logic to separate pure function:**
Take a look at the typescript function in the Angular component comment
Let's place it at ```src/utils/isOverlapRecurrentAppointment.js```

**2) Let's simplify the widget code:**
```suggestion
const handleAppointmentActions = (event, newAppointment) => {
        const recurrentAppointments = defaultData.filter((appointment) => appointment?.recurrenceRule)
            .map((appointment) => ({
                ...appointment,
                startDate: new Date(appointment.startDate),
                endDate: new Date(appointment.endDate),
            }));

        for (const recurrentAppointment of recurrentAppointments) {
            const isOverlap = isOverlapRecurrentAppointment(event, recurrentAppointment, newAppointment);

            if (isOverlap) {
                event.cancel = true;
                popup.show();
            }
        }
    };
```